src/iframeMessenger.js

```diff
@@ -423,6 +423,51 @@
             }
         }
 
+        function _addAcquisitionDataToLinks(acquisitionData) {
+            var links = document.getElementsByClassName('js-acquisition-link');
+            var json = encodeURIComponent(JSON.stringify(acquisitionData));
+            for (var i = 0; i < links.length; i++) {
+                var href = link.attr('href');
+                if (href) {
+                    href = `${href}${href.includes('?') ? '&' : '?'}${json}`;
+                    link.attr('href', href);                    
+                }
+            }
+        }
+
+        function _buildAcquisitionData(acquisitionData, referrerData) {
+            if (acquisitionData && 
+                typeof acquisitionData === 'object' && 
```

Comment:
you don't actually need the `typeof` check, if `acquisitionData` is truthy then object property access won't throw an error (`null` and `undefined` are the only values that will actually throw an error if you try to access a property on them. Strings, booleans & numbers all get automatically wrapped in object types)

Reply:
interesting. thanks.

Reply:
that's why you'll see stuff like

```javascript
var componentId = acquisitionsData && acquisitionsData.componentId
```